packages/rs-drive-abci/src/execution/quorum.rs

```diff
@@ -50,65 +50,39 @@ impl From<Quorum> for ValidatorSetUpdate {
                         node_ip,
                         platform_p2p_port
                     );
+
                     abci::ValidatorUpdate {
                         pub_key: Some(crypto::PublicKey {
                             sum: Some(Bls12381(public_key.to_bytes().to_vec())),
                         }),
                         power: 100,
-                        pro_tx_hash: pro_tx_hash.to_vec(),
+                        pro_tx_hash: reverse(&pro_tx_hash),
                         node_address,
                     }
                 })
                 .collect(),
             threshold_public_key: Some(crypto::PublicKey {
                 sum: Some(Bls12381(threshold_public_key.to_bytes().to_vec())),
             }),
-            quorum_hash: quorum_hash.to_vec(),
+            quorum_hash: reverse(&quorum_hash),
         }
     }
 }
 
+/// Reverse bytes
+///
+/// TODO: This is a workaround for reversed data returned by dashcore_rpc (little endian / big endian handling issue).
+/// We need to decide on a consistent approach to endianness and follow it.
+fn reverse(data: &[u8]) -> Vec<u8> {
+    let mut data = data.to_vec();
+    data.reverse();
+
+    data
+}
+
 impl From<&Quorum> for ValidatorSetUpdate {
     fn from(value: &Quorum) -> Self {
-        let Quorum {
-            quorum_hash,
-            validator_set,
-            threshold_public_key,
-            ..
-        } = value;
-        ValidatorSetUpdate {
-            validator_updates: validator_set
-                .iter()
-                .map(|(_, validator)| {
-                    let Validator {
-                        pro_tx_hash,
-                        public_key,
-                        node_ip,
-                        node_id,
-                        platform_p2p_port,
-                        ..
-                    } = validator;
-                    let node_address = format!(
-                        "tcp://{}@{}:{}",
-                        hex::encode(node_id.into_inner()),
-                        node_ip,
-                        platform_p2p_port
-                    );
-                    abci::ValidatorUpdate {
-                        pub_key: Some(crypto::PublicKey {
-                            sum: Some(Bls12381(public_key.to_bytes().to_vec())),
-                        }),
-                        power: 100,
-                        pro_tx_hash: pro_tx_hash.to_vec(),
-                        node_address: node_address.clone(),
-                    }
-                })
-                .collect(),
-            threshold_public_key: Some(crypto::PublicKey {
-                sum: Some(Bls12381(threshold_public_key.to_bytes().to_vec())),
-            }),
-            quorum_hash: quorum_hash.to_vec(),
-        }
+        value.clone().into()
```

Comment:
No, this is this not done on purpose. Cloning the entire quorum information is a lot more expensive.